pkg/k8s/statedb.go

```diff
@@ -51,6 +53,38 @@ func RegisterReflector[Obj any](jobGroup job.Group, db *statedb.DB, cfg Reflecto
 	return nil
 }
 
+// OnDemandTable provides an "on-demand" table of Kubernetes-derived objects.
+// The table is not populated until it is first acquired.
+// If the table should be cleared (to avoid holding onto the objects) when last reference
+// is released, set [ReflectorConfig.ClearTableOnStop].
+//
+// Intended to be used with [cell.Provide].
+// See [ExampleOnDemand] for example usage.
+func OnDemandTable[Obj any](jobs job.Registry, health cell.Health, log *slog.Logger, db *statedb.DB, cfg ReflectorConfig[Obj]) (hive.OnDemand[statedb.Table[Obj]], error) {
+	// Job group for the reflector that will be started when the table
+	// is acquired.
+	jg := jobs.NewGroup(
+		health,
+		job.WithLogger(log),
+	)
+
+	// If no ListerWatcher is given we assume that reflection is not wanted.
+	if cfg.ListerWatcher == nil {
```

Comment:
It is mandatory in the sense you need to decide on it. It can be nil if `client.Clientset` is disabled for example. Or if the feature that wants this is disabled. Having this avoids having to wrap the call to `RegisterReflector`.

Reply:
Ah I think I'll just remove this and require the wrapping. Much more obvious.